liquigraph-cli/src/main/java/org/liquigraph/cli/LiquigraphCli.java

```diff
@@ -105,11 +108,19 @@
     )
     private String dryRunOutputDirectory;
 
-
+    @Parameter(description = "[action] \n" +
+            "  action: Action to be executed. " +
+            "Accepted values are: " +
+            "run (default if --dry-run-output-directory is not set), " +
+            "dry-run (default if --dry-run-output-directory is set), " +
+            "clear-checksum"
+    )
+    private List<String> parameters = new ArrayList<>();
```

Comment:
If I understand the JCommander docs correctly, you can have a main parameter (i.e. without `names` attribute like you did above) that can be defined as of type`String`.
I assume JCommander will parse the action correctly and the custom parsing you do in the `action` method will not be necessary anymore (or at least drastically reduced).

Reply:
You can even implement a [custom converter](https://jcommander.org/#_custom_types_converters_and_splitters) so that the field is of type `Action`.

Reply:
> If I understand the JCommander docs correctly, you can have a main parameter

The issue is that JCommander sends all unknown parameters into parameters

i.e. when I use the following parameters

```
--changelog "../../../../cof-server-v2/infrastructure/neo4j-adapter/neo4j-adapter-migration/src/main/resources/db/liquigraph/changelog.xml" --username neo4j --password "toto" --graph-db-uri jdbc:neo4j:bolt://localhost:7688/ --unkown dry-run
```

I will have parameters (size 2) containing dry-run and --unkown.

What I can do is just check that parameters size is less or equals to 1 (and remove the extra filtering on --<option> I'm doing).

Reply:
Also, if I add arity = 1 to parameters, I'll have the following error message

```
com.beust.jcommander.ParameterException: There should be exactly 1 main parameters but 2 were found
```

Which is not very helpfull for the user since JCommander doesn't tell the user the name of the parameter in cause.

That's why I added code to check the length of parameters array

Reply:
Ok, I tested with the converter approach.

I'll create a converter that throws a ParameterException if the argument is not a valid action.

i.e. 
```
Exception in thread "main" com.beust.jcommander.ParameterException: Parameter action '--unkown' is invalid, must be one of DRY_RUN, RUN, CLEAR_CHECKSUM
```

Reply:
done in commit eee1086c